src/iceberg/util/string_util.h

```diff
@@ -44,6 +44,12 @@ class ICEBERG_EXPORT StringUtils {
                    [](char c) { return std::toupper(c); });    // NOLINT
     return input;
   }
+
+  static bool EqualsIgnoreCase(const std::string& a, const std::string& b) {
+    return a.size() == b.size() && std::ranges::equal(a, b, [](char ca, char cb) {
```

Comment:
`std::ranges::equal` will compare length so we don't need to write `a.size() == b.size()`